src/core/balance-sheet.ts

```diff
@@ -1,29 +1,55 @@
 import { getBalance as getBitcoindBalance } from "@services/bitcoind"
 import { lndsBalances } from "@services/lnd/utils"
 import { baseLogger } from "@services/logger"
-import { User } from "@services/mongoose/schema"
+import { InvoiceUser, User } from "@services/mongoose/schema"
 import { ledger } from "@services/mongodb"
 
 import { WalletFactory } from "./wallet-factory"
 
 const logger = baseLogger.child({ module: "balanceSheet" })
 
+export const updatePendingLightningTransactions = async () => {
+  let user, userWallet
+
+  // select distinct user ids from pending invoices
+  const usersWithPendingInvoices = InvoiceUser.aggregate([
+    { $match: { paid: false } },
+    { $group: { _id: "$uid" } },
+  ])
+    .cursor({ batchSize: 100 })
+    .exec()
+
+  for await (const { _id } of usersWithPendingInvoices) {
+    logger.trace("updating pending invoices for user %o", _id)
+    user = await User.findOne({ _id })
+    userWallet = await WalletFactory({ user, logger })
+    await userWallet.updatePendingInvoices()
+  }
+
+  const accounts = ledger.getAccountsWithPendingTransactions({ type: "payment" })
```

Comment:
`ledger.getAccountsWithPendingTransactions` is a [generator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*), internally use a cursor